### ISSUE TITLE
target-size must check whether nearby elements are obscured

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#4387]
The "Accept all cookies" link should not fail. The nearby "go to top" link is obscured:

```html
<div style="position: fixed; bottom: 0">
  <a href="#">Go to top</a>
</div>
<div style="position: fixed; bottom: 0; left: 0; right: 0; background: #eee">
  Cookies: <a href="#">Accept all cookies</a>
</div>
```

**Note** There are open issues related to this. Review them and keep them in mind when working on this.

**Comment by padmavemulapati** [comment]
Verified with the latest axe-core-develop branch code base,
`target-size` rule ignores and obscured for the elements nearby , findings can be in passes


![image](https://github.com/dequelabs/axe-core/assets/47409579/7bda1a25-80ba-4486-a74f-ea20ba4d1f5c)
